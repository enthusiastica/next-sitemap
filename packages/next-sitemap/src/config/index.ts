import fs from 'fs'
import allPath from '../path'
import { IConfig } from '../interface'
import deepmerge from 'deepmerge'

export const defaultConfig: Partial<IConfig> = {
  rootDir: 'public',
  priority: 0.7,
  changefreq: 'daily',
  sitemapSize: 5000,
  robotsTxtOptions: {
    policies: [
      {
        userAgent: '*',
        allow: '/'
      }
    ],
    additionalSitemaps: []
  }
}

export const withDefaultConfig = (config: IConfig) => deepmerge(defaultConfig, config)

export const loadConfig = (): IConfig => {
  if (fs.existsSync(allPath.CONFIG_FILE)) {
    const config = require(allPath.CONFIG_FILE)
    return withDefaultConfig(config)
  }

  throw new Error("No config file exist. Please create 'next-sitemap.js'")
}
