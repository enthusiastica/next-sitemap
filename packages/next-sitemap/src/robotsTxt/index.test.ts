import { generateRobotsTxt } from './index'
import { sampleConfig } from '../fixtures/config'

describe('next-sitemap/generateRobotsTxt', () => {
  test('generateRobotsTxt: generateRobotsTxt false in config', () => {
    expect(
      generateRobotsTxt({
        ...sampleConfig,
        generateRobotsTxt: false,
      } as any)
    ).toBeNull()
  })

  test('generateRobotsTxt: additionalSitemap', () => {
    expect(generateRobotsTxt(sampleConfig)).toMatchSnapshot()
  })
})
