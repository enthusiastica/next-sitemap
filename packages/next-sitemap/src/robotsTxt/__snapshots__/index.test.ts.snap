// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`next-sitemap/generateRobotsTxt generateRobotsTxt: additionalSitemap 1`] = `
"User-agent: *
Allow: /
User-agent: black-listed-bot
Disallow: /sub-path-1
Disallow: /path-2
Host: https://example.com
Sitemap: https://example.com/my-custom-sitemap-1.xml
Sitemap: https://example.com/my-custom-sitemap-2.xml
Sitemap: https://example.com/my-custom-sitemap-3.xml
"
`;
